scipy/stats/_discrete_distns.py

```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
 zipf = zipf_gen(a=1, name='zipf', longname='A Zipf')
 
 
+def _gen_harmonic(n, a=1):
+    """Generalized harmonic number"""
+    # See https://en.wikipedia.org/wiki/Harmonic_number; search for "hurwitz"
+    return zeta(a, 1) - zeta(a, n+1)
+
+
+class zipfian_gen(rv_discrete):
+    r"""A Zipfian discrete random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability mass function for `zipfian` is:
+
+    .. math::
+
+        f(k, a, n) = \frac{1}{H_{n,a} k^a}
+
+    for :math:`k \ge 1`, :math:`a \ge 0`, :math:`n \in {1, 2, 3, \dots}`.
+
+    `zipfian` takes :math:`a` and :math:`n` as shape parameters.
+    :math:`H_{n,a}` is the :math:`n^{\mbox{th}}` generalized harmonic number
+    of order :math:`a`.
+
+    The Zipfian distribution converges to the Zipf (zeta) distribution as
+    :math:`n \rightarrow \infty`.
+
+    %(after_notes)s
+
+    References
+    ----------
+    "Zipf's Law", Wikipedia, https://en.wikipedia.org/wiki/Zipf's_law
+
+    %(example)s
+
+    """
+    def _argcheck(self, a, n):
+        return (a >= 0) & (np.asarray(n) == np.asarray(n, dtype=int))
+
+    def _pmf(self, k, a, n):
+        Pk = 1.0 / _gen_harmonic(n, a) / k**a
+        return Pk
+
+    def _cdf(self, k, a, n):
+        return  _gen_harmonic(k, a)/_gen_harmonic(n, a)
+
+    def _sf(self, k, a, n):
+        k += 1  # # to match SciPy convention
+        # see http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
+        # implemented exactly as shown; please suggest improvements
+        return ((k**a*(_gen_harmonic(n, a) - _gen_harmonic(k, a)) + 1)
+                / (k**a*_gen_harmonic(n, a)))
+
+    def _stats(self, a, n):
+        # see # see http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
+        Hna = _gen_harmonic(n, a)
+        Hna1 = _gen_harmonic(n, a-1)
+        Hna2 = _gen_harmonic(n, a-2)
+        Hna3 = _gen_harmonic(n, a-3)
+        Hna4 = _gen_harmonic(n, a-4)
+        mu1 = Hna1/Hna
+        mu2n = (Hna2*Hna - Hna1**2)
+        mu2d = Hna**2
+        mu2 = mu2n / mu2d
+        g1 = (Hna3/Hna - 3*Hna1*Hna2/Hna**2 + 2*Hna1**3/Hna**3)/mu2**(3/2)
+        g2 = (Hna**3*Hna4 - 4*Hna**2*Hna1*Hna3 + 6*Hna*Hna1**2*Hna2
+              - 3*Hna1**4) / mu2n**2 - 3
```

Comment:
```suggestion
              - 3*Hna1**4) / mu2n**2 - 3
```
Why is subtracting 3 needed? I see this in `dlaplace_gen` and `betabinom_gen`, and that's exactly what was needed here.